src/main/java/net/minecraftforge/items/wrapper/SidedInvWrapper.java

```diff
@@ -0,0 +1,197 @@
+package net.minecraftforge.items.wrapper;
+
+import net.minecraft.inventory.ISidedInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemHandlerHelper;
+
+public class SidedInvWrapper implements IItemHandler {
+    protected final ISidedInventory inv;
+    protected final EnumFacing side;
+
+    public SidedInvWrapper(ISidedInventory inv, EnumFacing side)
+    {
+        this.inv = inv;
+        this.side = side;
+    }
+
+    public static int getSlots(ISidedInventory inv, EnumFacing side)
```

Comment:
Why are all these helper methods public static? All this information should be cached int he constructor locally?